lib/new-admin/routes/auth.js

```diff
@@ -0,0 +1,259 @@
+const otplib = require('otplib')
+const bcrypt = require('bcrypt')
+
+const users = require('../../users')
+const login = require('../login')
+
+async function isValidUser (username, password) {
+  const hashedPassword = await login.checkUser(username)
+  if (!hashedPassword) return false
+
+  const isMatch = await bcrypt.compare(password, hashedPassword)
+  if (!isMatch) return false
+
+  const user = await login.validateUser(username, hashedPassword)
+  if (!user) return false
+  return user
+}
+
+module.exports = function (app) {
+  app.post('/api/login', function (req, res, next) {
+    const usernameInput = req.body.username
+    const passwordInput = req.body.password
+
+    isValidUser(usernameInput, passwordInput).then(user => {
+      if (!user) return res.sendStatus(403)
+      users.get2FASecret(user.id).then(user => {
+        const twoFASecret = user.twofa_code
+        if (twoFASecret) return res.status(200).json({ message: 'INPUT2FA' })
+        if (!twoFASecret) return res.status(200).json({ message: 'SETUP2FA' })
+      })
+    })
+  })
+
+  app.post('/api/login/2fa', function (req, res, next) {
+    const code = req.body.twoFACode
+    const username = req.body.username
+    const password = req.body.password
+    const rememberMeInput = req.body.rememberMe
+
+    isValidUser(username, password).then(user => {
+      if (!user) return res.sendStatus(403)
+
+      users.get2FASecret(user.id).then(user => {
+        const secret = user.twofa_code
+        const isCodeValid = otplib.authenticator.verify({ token: code, secret: secret })
+        if (!isCodeValid) return res.sendStatus(403)
+
+        const finalUser = { id: user.id, username: user.username, role: user.role }
+        req.session.user = finalUser
+        if (rememberMeInput) req.session.cookie.maxAge = 90 * 24 * 60 * 60 * 1000 // 90 days
```

Comment:
we have a time.js file for this, also create a variable near the top of the file for this

`const REMEMBER_ME_AGE = `